### ISSUE TITLE
Feature list: Can't use keyword search from feature list page

### ISSUE THREAD

**Issue by mshukla1** [PATRIC3/patric3_website#1801]
Can't use keyword search from feature list page
-----------------------------------------------

               Key: PATRIC-1152
               URL: https://jira.cels.anl.gov/browse/PATRIC-1152
           Project: PATRIC
        Issue Type: Bug
          Reporter: kcherny1@binghamton.edu
          Assignee: Rebecca Wattam

Version: 3.4.13 
URL: https://www.patricbrc.org/view/FeatureList/#view_tab=features 
User ID: kcherny1
I have my feature list, when I use the keyword search from the feature list 
page, it searches across all features, 
instead of within my feature list. 

-Maulik for user.

**Comment by dawenx** [comment]
I think this is addressed in #1779 in which I enabled "Public" filter on feature tab. The fix is currently on alpha, once it's pushed to the production, the user can use this filter to narrow the list to private features, then (s)he can perform a keyword search.

**Comment by dawenx** [comment]
i spoke too soon, the fix is not on alpha yet. but i submitted a PR on 11/15. @hyoo

**Comment by hyoo** [comment]
Here is what i understood. 
Lets say user has some selected features in the features list,

https://www.patricbrc.org/view/FeatureList/?in(feature_id,(PATRIC.1094551.3.AIME01000001.CDS.30293.30463.fwd,PATRIC.1094551.3.AIME01000001.CDS.28639.28776.fwd,PATRIC.1094551.3.AIME01000001.CDS.27108.28634.rev,PATRIC.1094551.3.AIME01000001.CDS.25960.26895.rev,PATRIC.1094551.3.AIME01000001.CDS.25405.25896.rev,PATRIC.1094551.3.AIME01000001.CDS.24771.25349.rev,PATRIC.1094551.3.AIME01000001.CDS.24114.24449.fwd,PATRIC.1094551.3.AIME01000001.CDS.21783.23426.fwd,PATRIC.1094551.3.AIME01000001.CDS.21408.21704.fwd,PATRIC.1094551.3.AIME01000001.CDS.19806.21194.rev,PATRIC.1094551.3.AIME01000001.CDS.19596.19712.rev,PATRIC.1094551.3.AIME01000001.CDS.19125.19259.fwd,PATRIC.1094551.3.AIME01000001.CDS.18657.18785.fwd,PATRIC.1094551.3.AIME01000001.CDS.16987.18612.fwd,PATRIC.1094551.3.AIME01000001.CDS.16260.16784.rev,PATRIC.1094551.3.AIME01000001.CDS.15681.15884.fwd,PATRIC.1094551.3.AIME01000001.CDS.15513.15680.fwd,PATRIC.1094551.3.AIME01000001.CDS.13303.14796.fwd,PATRIC.1094551.3.AIME01000001.CDS.13045.13164.fwd,PATRIC.1094551.3.AIME01000001.CDS.12015.12191.rev))#view_tab=features&filter=and(or(eq(feature_type,%22CDS%22)),or(eq(annotation,%22PATRIC%22)),false)

This is selected from Taxon level features tab. 

and when I add extra keyword to search, it works as expected. 

Do you know what was the selection?

**Comment by dawenx** [comment]
sorry, i don't quite understand this issue. i thought adding 'public' field to the filter solves this issue already. @mshukla1 can you explain a little bit if my understanding is not correct?

**Comment by mshukla1** [comment]
@dawenx 

I figured out how to reproduce this problem. GO to the following page. Select all features, and then click on the feature list from the action bar. It will take you to the feature list page, with feature ids passed as post method, not get. If you use keyword search from that page, it results in searching across all features in the database, not just those in he feature list. 

https://www.patricbrc.org/view/Genome/83332.12#view_tab=features

I can show it to you on skype Tomorrow. 

-Maulik

**Comment by dawenx** [comment]
i can reproduce it, will take a look.

**Comment by dawenx** [comment]
i discussed this with @dmachi last week. this seems to be a tricky issue that requires work on p3app.js. @dmachi will take a look.

**Comment by mshukla1** [comment]
@dmachi @dawenx 

any update on this?

**Comment by dawenx** [comment]
I will check with @dmachi to see what we can do on this.

**Comment by dmachi** [comment]
I have not had a chance to look into this yet outside of the discussions I had with Dawen previously.

**Comment by JoshuaVSherman** [comment]
I was able to reproduce it, and discussed with Dustin. Since the url string no longer contains the query (because the length of it was too long for the get request), it uses a post request to build the Feature List table, and then the resulting url no longer has any query string to build off of for keyword. I have tried to isolate where the post is being created, possibly we could refactor this to always do a post rather than a get and  store the body of the post in a variable to be used by and modified in subsequent post requests. My other idea was to remove the keyword filter from the query altogether and just filter the table directly in the front-end. Either of these ideas require considerable effort, so I agree that this is something for Dustin, not me.
